src/adapter/src/session.rs

```diff
@@ -1084,6 +1097,23 @@ impl<T: TimestampManipulation> TransactionStatus<T> {
                     TransactionOps::SingleStatement { .. } => {
                         return Err(AdapterError::SingleStatementTransaction)
                     }
+                    TransactionOps::DDL {
+                        ops: og_ops,
+                        revision: og_revision,
+                        state: _,
+                    } => match add_ops {
+                        TransactionOps::DDL {
+                            ops: new_ops,
+                            revision: new_revision,
+                            state: _,
+                        } => {
+                            if *og_revision != new_revision {
+                                return Err(AdapterError::DDLTransactionRace);
+                            }
+                            og_ops.extend(new_ops);
```

Comment:
It is odd that ops and state might be out of sync here, and this is at odds with the doc comments. That seems bad because users of this will need to be aware of that and maintain invariants outside of this function. Perhaps the new state should also be passed to add_ops to maintain the property that ops and state always match.

Reply:
Yeah I didn't love this control flow. I reworked it a bit so the only place `ops` and `state` should get modified is in this method! I'm happier with it now, let me know what you think

Reply:
> the only place ops and state should get modified is in this method

The current code still doesn't do this.

Reply:
Ah geez, sorry about the back and forth here. I made the changes but I must have not pushed them. The only place we mutate the `ops` and `state` in a transaction should now be in `fn add_ops(...)`